### ISSUE TITLE
Error: unexpected end of json input

### ISSUE THREAD

**Issue by stevez** [cenfun/monocart-reporter#95]
When tried to use 'inputDir', and I got following error:

Error in reporter SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at addJsonData (/node_modules/monocart-coverage-reports/lib/generate.js:472:35)
    at getInputData (/node_modules/monocart-coverage-reports/lib/generate.js:490:17)
    at CoverageReport.generate (/node_modules/monocart-coverage-reports/lib/index.js:104:43)
    at generateGlobalCoverageReport (/Users/

What I can confirm:
1) neither raw json nor non raw json file will get the same error
2) I double check the json file with online json editor, they are all valid
3) I even tried small sized json generated by jest, still not working 

Could you provide an working example which has both jest tests and playwright tests, and config for the merging coverage?

**Comment by cenfun** [comment]
Sorry, I can't provide an example because Jest + Playwright has too many possible options.
It seems you using `inputDir` option of `monocart-coverage-reports` in the playwright custom reporter `monocart-reporter`, I haven't tested this situation yet. Who provides your raw json? It requires `raw` report.

So, what is your requirement and workflow?
1, Jest + Playwright? Jest for unit test and Playwright for e2e test?
2, nextjs? coverage both server side and client side?
3, V8 or istanbul? We can NOT merge v8 with istanbul, so only one can be chosen.

Guess the possible workflow:
1, Jest + [jest-monocart-coverage](https://github.com/cenfun/jest-monocart-coverage) output raw coverage data with `raw` report `./ccoverage-reports/unit/raw`
2, Playwright + `monocart-coverage-reports` output raw coverage data with `raw` report `./ccoverage-reports/e2e/raw`
3, Merge raw coverage data with `inputDir` option of `monocart-coverage-reports`, create a script and run it after step1 and step2

Or provide a small repo which I can reproduce the json error locally, i could debug and fix it.

**Comment by stevez** [comment]
1) test coverage is using Istanbul
2) I did use your jest Monocart report plugin to generate the raw files
3) it is not Nextjs, we used custom express server and webpack library implemented the ssr
4)e2e uses playwright

I am trying to understand the workflow 
Right now I only put the jest raw folder in the inputDir, and output folder is the merged folder

do I need to put the e2e raw folder as well?

**Comment by cenfun** [comment]
Your workflow seems no problem.
I guess the issue might be because you used the Istanbul format, which I have not tested, as I prefer V8. let me test it locally.

**Comment by cenfun** [comment]
In fact, I have tested merging Istanbul raw json reports locally and it works.
It also need you to provide more help so that I can reproduce the problem.

**Comment by stevez** [comment]
Good news, after I upgraded the latest version, it is working now, but I did found out the cache folder caused the problem: even I made changes, I still got the same errors, after I delete the entire folder, now it is working.

But I found another issue: when running jest tests using your jest monocart report plugin, there is no output in the console, only the coverage report will be shown at the end. I am pretty sure the plugin caused the problem, since when I commented out the monocart report in the jest config, the test messages will shown again.

**Comment by cenfun** [comment]
the test meesage is requires a report, for example `text`, please check your config
```
coverageReporters: ['none'], // remove this none report
```

**Comment by stevez** [comment]
I tried removing the 'none', but still not working: what I mean is the messages jest showing the test and progress in the console are missing, the coverage reports are ok, they are showing at the end of the tests

**Comment by stevez** [comment]
I got it fixed, just adding 'default' in the reports array.

Based on jest doc:
https://jestjs.io/docs/configuration#reporters-arraymodulename--modulename-options

Default Reporter

If custom reporters are specified, the default Jest reporter will be overridden. If you wish to keep it, 'default' must be passed as a reporters name:

JavaScript
TypeScript
/** @type {import('jest').Config} */
const config = {
  reporters: [
    'default',
    ['jest-junit', {outputDirectory: 'reports', outputName: 'report.xml'}],
  ],
};

module.exports = config;

Now everything works, you can close this issue, but please update the doc for this issue

**Comment by cenfun** [comment]
Great, thanks for the fixing, I will update the docs.
